module/behaviour/skills/HeadBehaviourSoccer/src/HeadBehaviourSoccer.cpp

```diff
@@ -132,546 +88,58 @@ namespace module::behaviour::skills {
-                fixationObjects.balls.push_back(ball);
-            }
-            else {
-                search = true;
+        // Updates the last seen time of ball
+        on<Trigger<VisionBalls>, With<Sensors>>().then([this](const VisionBalls& balls, const Sensors& sensors) {
+            if (!balls.balls.empty()) {
+                ballLastMeasured = NUClear::clock::now();
+                rBCc = Eigen::Vector3d(sphericalToCartesian(balls.balls[0].measurements[0].srBCc.cast<double>()));
             }
-        }
-
-        return fixationObjects;
-    }
-
-    Goals HeadBehaviourSoccer::getFixationObjects(const std::shared_ptr<const Goals>& vgoals, bool& search) {
-
-        auto now = NUClear::clock::now();
-        Goals fixationObjects;
-
-        int maxPriority = std::max(std::max(ballPriority, goalPriority), 0);
-        if (ballPriority == goalPriority) {
-            log<NUClear::WARN>("HeadBehaviourSoccer - Multiple object searching currently not supported properly.");
-        }
-
-        // TODO(BehaviourTeam): make this a loop over a list of objects or something
-        // Get goals
-        if (goalPriority == maxPriority) {
-            if (vgoals && !vgoals->goals.empty()) {
-                // Fixate on goals and lines and other landmarks
-                timeLastObjectSeen = now;
-                std::set<Goal::Side> visiblePosts;
-                // TODO(BehaviourTeam): treat goals as one object
-                Goals goals;
-                for (const auto& goal : vgoals->goals) {
-                    visiblePosts.insert(goal.side);
-                    goals.goals.push_back(goal);
+        });
+
+        on<Trigger<Sensors>, Single, Sync<HeadBehaviourSoccer>>().then(
+            "Head Behaviour Main Loop",
+            [this](const Sensors& sensors) {
+                // Get the time since the ball was last seen
+                float timeSinceBallLastMeasured =
+                    std::chrono::duration_cast<std::chrono::duration<float>>(NUClear::clock::now() - ballLastMeasured)
+                        .count();
+                // Only look for ball if not getting up
+                if (!isGettingUp) {
+                    if (timeSinceBallLastMeasured < search_timeout_ms / 1000) {
+                        // We can see the ball, lets look at it
+                        Eigen::Vector2d angles               = screenAngularFromObjectDirection(rBCc);
+                        std::unique_ptr<HeadCommand> command = std::make_unique<HeadCommand>();
+                        command->yaw                         = angles[0];
+                        command->pitch                       = angles[1];
+                        command->robot_space                 = true;
+                        command->smooth                      = true;
+                        emit(std::move(command));
+                    }
+                    else {
+                        // We haven't seen the ball in a while, lets look around
```

Comment:
Maybe have something more specific, like
'Balll hasn't been seen in a while. Look around using search positions.'